contracts/DVA/DVATransferManager.sol

```diff
@@ -0,0 +1,651 @@
+     *  @param includeRecipientApprover determines whether the recipient is included in the approver list
+     *  @param includeAgentApprover determines whether the agent is included in the approver list
+     *  @param sequentialApproval determines whether approvals must be sequential
+     *  @param additionalApprovers are the addresses of additional approvers to be added to the approver list
+     *  Only an agent of a token can call this function
+     *  DVATransferManager must be an agent of the given token
+     *  emits an `ApprovalCriteriaModified` event
+     */
+    function modifyApprovalCriteria(
+        address tokenAddress,
+        bool includeRecipientApprover,
+        bool includeAgentApprover,
+        bool sequentialApproval,
+        address[] memory additionalApprovers
+    ) external {
+        if (!AgentRole(tokenAddress).isAgent(msg.sender)) {
+            revert OnlyTokenAgentCanCall(tokenAddress);
+        }
+
+        if (!IToken(tokenAddress).identityRegistry().isVerified(address(this))) {
+            revert DVAManagerIsNotVerifiedForTheToken(tokenAddress);
+        }
+
+        bytes32 hash = keccak256(
+            abi.encode(
+                tokenAddress,
+                includeRecipientApprover,
+                includeAgentApprover,
+                additionalApprovers
+            )
+        );
+
+        _approvalCriteria[tokenAddress] = ApprovalCriteria(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash);
+
+        emit ApprovalCriteriaModified(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash
+        );
+    }
+
+    /**
+     *  @dev initiates a new transfer
+     *  @param tokenAddress is the address of the token
+     *  @param recipient is the address of the recipient
+     *  @param amount is the transfer amount
+     *  Approval criteria must be preset for the given token address
+     *  Receiver must be verified for the given token address
+     *  emits a `TransferInitiated` event
+     */
+    function initiateTransfer(address tokenAddress, address recipient, uint256 amount) external {
+        ApprovalCriteria memory approvalCriteria = _approvalCriteria[tokenAddress];
+        if (approvalCriteria.tokenAddress == address(0)) {
+            revert TokenIsNotRegistered(tokenAddress);
+        }
+
+        IToken token = IToken(tokenAddress);
+        if (!token.identityRegistry().isVerified(recipient)) {
+            revert RecipientIsNotVerified(tokenAddress, recipient);
+        }
+
+        bool transferSent = token.transferFrom(msg.sender, address(this), amount);
+        if (!transferSent) {
+            revert TokenTransferFailed(tokenAddress, msg.sender, address(this), amount);
```

Comment:
I don't think this code is reachable. When transfer fails it reverts, therefore i don't see how we could get the case where `transferFrom` returns `false` and reach this code. I would therefore remove this part and the custom error that goes with it